mono/tests/threadpool-exceptions5.cs

```diff
@@ -2,37 +2,47 @@
 using System.Threading;
 
 class Test {
-	static object monitor;
-	static int return_value = 2;
+
+	static ManualResetEvent mre = new ManualResetEvent (false);
+
 	static int Main ()
 	{
-		monitor = new object ();
 		AppDomain.CurrentDomain.UnhandledException += OnUnhandledException;
 		WaitCallback wcb = new WaitCallback ((a) => {
 			Thread.CurrentThread.Abort();
 		});
+
 		wcb.BeginInvoke (wcb, OnCBFinished, null);
-		lock (monitor) {
-			Monitor.Wait (monitor);
-		}
-		Thread.Sleep (1000);
-		return 1;
+
+		if (!mre.WaitOne (10000))
+			return 2;
+
+		GC.Collect ();
```

Comment:
What's the need for these 2 new function calls?

Reply:
same as above, could remove then but they help to ensure there is no other side effect